src/Forms/HTMLEditor/HTMLEditorConfig.php

```diff
@@ -197,46 +333,191 @@ public static function get_available_configs_map()
+    {
+        if (empty($rulesArray)) {
+            $this->setElementRuleSet(new HTMLEditorRuleSet());
+            return $this;
+        }
+        if (!ArrayLib::is_associative($rulesArray)) {
+            throw new InvalidArgumentException('Element rules array must be associative.');
+        }
+
+        $ruleset = new HTMLEditorRuleSet();
+
+        // The global rule is a bit special
+        if (isset($rulesArray[HTMLEditorElementRule::GLOBAL_NAME])) {
+            $globalRuleDefinition = $rulesArray[HTMLEditorElementRule::GLOBAL_NAME];
+            // Only the 'attributes' key is allowed for the global rule.
+            if (!empty(array_diff(array_keys($globalRuleDefinition), ['attributes']))) {
+                throw new InvalidArgumentException(HTMLEditorElementRule::GLOBAL_NAME . ' element rule can only have attributes.');
+            }
+            // Add global rule attributes
+            $this->addAttributeRulesToElementRule($globalRuleDefinition['attributes'] ?? [], $ruleset->getGlobalRule());
+            unset($rulesArray[HTMLEditorElementRule::GLOBAL_NAME]);
+        }
+
+        // Add conversion rules first.
+        foreach ($rulesArray as $ruleName => $ruleDefinition) {
+            if (!is_array($ruleDefinition)) {
+                continue;
+            }
+            $convertTo = $ruleDefinition['convertTo'] ?? '';
+            // Add the conversion rule if there is one, and if the element to convert to is allowed.
+            if (!empty($convertTo) && isset($rulesArray[$convertTo]) && $rulesArray[$convertTo] !== false) {
+                $ruleset->addElementSubstitutionRule($ruleName, $convertTo);
+            }
+        }
+        // Add concrete element rules
+        foreach ($rulesArray as $ruleName => $ruleDefinition) {
+            // Allow rules to be disabled through overridden configuration.
+            // Also skip elements slated for conversion because we added those already.
+            if ($ruleDefinition === null || $ruleDefinition === false || isset($ruleDefinition['convertTo'])) {
+                continue;
+            }
+            // Use default rule definition if none is supplied
+            if ($ruleDefinition === true) {
+                $ruleDefinition = [];
+            }
+            $elementRule = HTMLEditorElementRule::fromArray($ruleName, $ruleDefinition);
+            if (isset($ruleDefinition['attributes'])) {
+                $this->addAttributeRulesToElementRule($ruleDefinition['attributes'], $elementRule);
+            }
+            $ruleset->addElementRule($elementRule);
+        }
+
+        return $this->setElementRuleSet($ruleset);
+    }
+
+    /**
+     * Given an associative array of attribute rule definitions, instantiate the HTMLEditorAttributeRules
+     * and add them to the given HTMLEditorElementRule
+     */
+    private function addAttributeRulesToElementRule(array $attributeRuleDefinitions, HTMLEditorElementRule $elementRule): void
```

Comment:
Move new private function below the public / protected functions

Reply:
Done